modules/widgets/recurring-payments.php

```diff
@@ -0,0 +1,179 @@
+			if ( ! empty( $title ) ) {
+				echo $args['before_title'] . $title . $args['after_title'];
+			}
+
+			echo '<div class="jetpack-recurring-payments-content">';
+			echo Jetpack_Memberships::get_instance()->render_button( array(
+				'planId' => $instance['product_post_id'],
+				'submitButtonText' => $instance['text'],
+			) );
+
+
+			echo '</div><!--simple-payments-->';
+
+			echo $args['after_widget'];
+
+			/** This action is already documented in modules/widgets/gravatar-profile.php */
+			do_action( 'jetpack_stats_extra', 'widget_view', 'recurring_payments' );
+		}
+
+		/**
+		 * Gets the latests field value from either the old instance or the new instance.
+		 *
+		 * @param array $mixed Array of values for the new form instance.
+		 * @param array $mixed Array of values for the old form instance.
+		 * @return mixed $mixed Field value.
+		 */
+		private function get_latest_field_value( $new_instance, $old_instance, $field ) {
+			return ! empty( $new_instance[ $field ] )
+				? sanitize_text_field( $new_instance[ $field ] )
+				: $old_instance[ $field ];
+		}
+
+		/**
+		 * Sanitize widget form values as they are saved.
+		 *
+		 * @see WP_Widget::update()
+		 *
+		 * @param array $new_instance Values just sent to be saved.
+		 * @param array $old_instance Previously saved values from database.
+		 *
+		 * @return array Updated safe values to be saved.
+		 */
+		function update( $new_instance, $old_instance ) {
+			$defaults = $this->defaults();
+			//do not overrite `product_post_id` for `$new_instance` with the defaults
+			$new_instance = wp_parse_args( $new_instance, array_diff_key( $defaults, array( 'product_post_id' => 0 ) ) );
+			$old_instance = wp_parse_args( $old_instance, $defaults );
+
+			return array(
+				'title'           => $this->get_latest_field_value( $new_instance, $old_instance, 'title' ),
+				'product_post_id' => $this->get_latest_field_value( $new_instance, $old_instance, 'product_post_id' ),
+				'text'     => $this->get_latest_field_value( $new_instance, $old_instance, 'text' ),
+			);
+
+		}
+
+		/**
+		 * Back-end widget form.
+		 *
+		 * @see WP_Widget::form()
+		 *
+		 * @param array $instance Previously saved values from database.
+		 */
+		function form( $instance ) {
+			$instance = wp_parse_args( $instance, $this->defaults() );
+			$product_posts = get_posts(
+				array(
+					'numberposts' => 100,
+					'orderby'     => 'date',
+					'post_type'   => Jetpack_Memberships::$post_type_plan,
+					'post_status' => 'publish',
+				)
+			);
+
+			$product_posts = array_map( function( $post ) {
+				return Jetpack_Memberships::product_post_to_array( $post );
+			}, $product_posts );
+			$blog_id = Jetpack_Memberships::get_blog_id();
+			require dirname( __FILE__ ) . '/recurring-payments/form.php';
+		}
+	}
+
+	// Register Jetpack_Recurring_Payments_Widget widget.
+	function register_widget_jetpack_recurring_payments() {
+		if ( ! class_exists( 'Jetpack_Memberships' ) ) {
```

Comment:
In general, I think it would be best to not register the widget at all if the site can't use Recurring Payments, a bit like what we do for Search maybe?
https://github.com/Automattic/jetpack/blob/e3986c699471dbb1dab42bdeac2041b8c979514c/modules/widgets/search.php#L13

Reply:
I have to actually add the feature to the plans objects.
It may be a time to do that - I'll focus on that now 👍

Reply:
Ok, done!